engine/src/main/java/org/terasology/config/flexible/constraints/SettingConstraint.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible.constraints;
+
+import org.terasology.config.flexible.Setting;
+
+/**
+ * Represents constraints on values that can be stored in a {@link Setting}.
+ *
+ * @param <T> The type of values that are constrained by this {@link SettingConstraint}.
```

Comment:
Huh does a `<T>` like this get javadocced as a param? Never actually thought about that before

Reply:
Yes, the type parameters can also be documented.

Reply:
Yup, and I actually really like the fact that you went the extra mile and documented it.